modules/tensor_mechanics/src/materials/cohesive_zone_model/SlipWeakeningFriction2d.C

```diff
@@ -0,0 +1,163 @@
+    _T2_o(getParam<Real>("T2_o")),
+    _mu_d(getParam<Real>("mu_d")),
+    _Dc(getParam<Real>("Dc")),
+    _len(getParam<Real>("len")),
+    _density(getMaterialPropertyByName<Real>(_base_name + "density")),
+    _rot(getMaterialPropertyByName<RankTwoTensor>(_base_name + "czm_total_rotation")),
+    _disp_slipweakening_x(coupledValue("disp_slipweakening_x")),
+    _disp_slipweakening_neighbor_x(coupledNeighborValue("disp_slipweakening_x")),
+    _disp_slipweakening_y(coupledValue("disp_slipweakening_y")),
+    _disp_slipweakening_neighbor_y(coupledNeighborValue("disp_slipweakening_y")),
+    _reaction_slipweakening_x(coupledValue("reaction_slipweakening_x")),
+    _reaction_slipweakening_neighbor_x(coupledNeighborValue("reaction_slipweakening_x")),
+    _reaction_slipweakening_y(coupledValue("reaction_slipweakening_y")),
+    _reaction_slipweakening_neighbor_y(coupledNeighborValue("reaction_slipweakening_y")),
+    _disp_slipweakening_x_old(coupledValueOld("disp_slipweakening_x")),
+    _disp_slipweakening_neighbor_x_old(coupledNeighborValueOld("disp_slipweakening_x")),
+    _disp_slipweakening_y_old(coupledValueOld("disp_slipweakening_y")),
+    _disp_slipweakening_neighbor_y_old(coupledNeighborValueOld("disp_slipweakening_y")),
+    _mu_s(coupledValue("mu_s")),
+    _ini_shear_sts(coupledValue("ini_shear_sts"))
+{
+}
+
+void
+SlipWeakeningFriction2d::computeInterfaceTractionAndDerivatives()
+{
+  // Global Displacement Jump
+  RealVectorValue displacement_jump_global(
+      _disp_slipweakening_x[_qp] - _disp_slipweakening_neighbor_x[_qp],
+      _disp_slipweakening_y[_qp] - _disp_slipweakening_neighbor_y[_qp]);
+
+  // Global Displacement Jump Old
+  RealVectorValue displacement_jump_old_global(
+      _disp_slipweakening_x_old[_qp] - _disp_slipweakening_neighbor_x_old[_qp],
+      _disp_slipweakening_y_old[_qp] - _disp_slipweakening_neighbor_y_old[_qp]);
+
+  // Global Displacement Jump Rate
+  RealVectorValue displacement_jump_rate_global =
+      (displacement_jump_global - displacement_jump_old_global) * (1 / _dt);
+
+  // Local Displacement Jump / Displacement Jump Rate
+  RealVectorValue displacement_jump = _rot[_qp].transpose() * displacement_jump_global;
+  RealVectorValue displacement_jump_rate = _rot[_qp].transpose() * displacement_jump_rate_global;
+
+  // Parameter initialization
+  Real mu_s = 0;
+  Real mu_d = _mu_d;
+  Real Dc = _Dc;
+  Real tau_f = 0;
+
+  Real T1_o = 0;
+  Real T2_o = _T2_o;
+
+  Real area = _len;
+
+  // Reaction force in local coordinate
+  RealVectorValue R_plus_global(
+      -_reaction_slipweakening_x[_qp], -_reaction_slipweakening_y[_qp], 0);
+  RealVectorValue R_minus_global(
+      -_reaction_slipweakening_neighbor_x[_qp], -_reaction_slipweakening_neighbor_y[_qp], 0);
+
+  RealVectorValue R_plus_local = _rot[_qp].transpose() * R_plus_global;
+  RealVectorValue R_minus_local = _rot[_qp].transpose() * R_minus_global;
+
+  Real R_plus_local_x = R_plus_local(1);
```

Comment:
Are the local coordinates based on the cohesive zone normals?  If so, then I think you should use something like n,t,s for normal, and then s and t for the tangential components of if there is a syntax that is commonly used in other cohesive zone models then you should use that.   @jiangwen84 do you know?

Reply:
yes I will make n,t,s instead of y,x,z, here I assume the global coordinate aligns with the local coordinate in this example but should be generalized